src/zenml/integrations/mlflow/model_registries/mlflow_model_registry.py

```diff
@@ -0,0 +1,796 @@
+            registered_model_version = self.mlflow_client.create_model_version(
+                name=name,
+                source=model_source_uri,
+                run_id=registry_metadata.get("mlflow_run_id")
+                if registry_metadata
+                else "",
+                run_link=registry_metadata.get("mlflow_run_link")
+                if registry_metadata
+                else "",
+                description=version_description,
+                tags=version_tags,
+            )
+        except MlflowException as e:
+            logger.error(
+                f"Failed to register new model version for model '{name}'."
+            )
+            raise e
+        # Return the registered model version.
+        return ModelVersion(
+            model_registration=ModelRegistration(
+                name=registered_model_version.name,
+                description=description,
+                tags=tags,
+            ),
+            model_source_uri=registered_model_version.source,
+            registry_metadata={
+                "mlflow_run_id": registered_model_version.run_id or "",
+                "mlflow_run_link": registered_model_version.run_link or "",
+            },
+            version=registered_model_version.version,
+            description=registered_model_version.description,
+            version_stage=registered_model_version.current_stage,
+            tags=registered_model_version.tags,
+            created_at=str(registered_model_version.creation_timestamp),
+            last_updated_at=str(
+                registered_model_version.last_updated_timestamp
+            ),
+        )
+
+    def delete_model_version(
+        self,
+        name: str,
+        version: str,
+    ) -> None:
+        """Delete a model version from the MLflow model registry.
+
+        Args:
+            name (str): The name of the model.
+            version (str): The version of the model.
+
+        Raises:
+            MlflowException: If mlflow fails to delete the model version.
+            KeyError: If the model version does not exist.
+        """
+        if not self.check_model_version_exists(name, version):
+            raise KeyError(
+                f"The model version with name '{name}' and version '{version}' "
+                "does not exist."
+            )
+        try:
+            self.mlflow_client.delete_model_version(
+                name=name,
+                version=version,
+            )
+        except MlflowException as e:
+            logger.error(
+                f"Failed to delete model version '{name}:{version}' from the "
+                "MLflow model registry."
+            )
+            raise e
+
+    def update_model_version(
+        self,
+        name: str,
+        version: str,
+        version_description: Optional[str] = None,
+        version_tags: Optional[Dict[str, str]] = None,
+        version_stage: Optional[ModelVersionStage] = None,
```

Comment:
```suggestion
        description: Optional[str] = None,
        tags: Optional[Dict[str, str]] = None,
        stage: Optional[ModelVersionStage] = None,
```

Doesn't it just look better ?